### PR TITLE
[GH-#369] Represent naming state using patricia merkle tree 

### LINKED ISSUE #369: Represent naming state using patricia merkle tree

### PR DESCRIPTION
**Pull request by DanielaIvanova**
resolves #369

### CODE REVIEW

File: apps/aecore/lib/aecore/naming/naming_state_tree.ex [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    PatriciaMerkleTree.new(:naming)
+  end
+
+  @spec put(namings_state(), binary(), Naming.t()) :: namings_state()
+  def put(trie, key, value) do
+    serialized = serialize(value)
+    PatriciaMerkleTree.enter(trie, key, serialized)
+  end
+
+  @spec get(namings_state(), binary()) :: Naming.t()
```
Comment (cytadela8) [inline apps/aecore/lib/aecore/naming/naming_state_tree.ex]: if you change to `@spec get(namings_state(), binary()) :: Naming.t() | :none` you should not get some of the suppressed dialyzer warnings.

Comment (DanielaIvanova) [reply apps/aecore/lib/aecore/naming/naming_state_tree.ex]: Done

**Comment by thepiwo** [comment]
does this include the changes from https://github.com/aeternity/elixir-node/pull/368?

**Comment by DanielaIvanova** [comment]
@thepiwo No everything

**Comment by thepiwo** [comment]
@DanielaIvanova what do you mean? so it includes changes from #368 or not?

**Comment by DanielaIvanova** [comment]
@thepiwo We made branch #369 from branch #366 but we added requested changes from PR #368 that means in #369 is not everything included from #368

**Comment by thepiwo** [comment]
thank you, but to merge as #368

**Comment by DanielaIvanova** [comment]
@thepiwo We will

**Comment by DanielaIvanova** [comment]
@thepiwo Done

**Comment by thepiwo** [comment]
so, we wait for the other PMT PR to be merged.

**Comment by thepiwo** [comment]
will close this until unblocked

**Pull request closed without merging**